autopilot/hostfilter.go

```diff
@@ -224,50 +224,51 @@ func isUsableHost(cfg api.AutopilotConfig, rs api.RedundancySettings, gc worker.
 func isUsableContract(cfg api.AutopilotConfig, ci contractInfo, bh uint64, renterFunds types.Currency, f *ipFilter) (usable, recoverable, refresh, renew bool, reasons []string) {
 	c, s := ci.contract, ci.settings
 
+	usable = true
 	if bh > c.EndHeight() {
 		reasons = append(reasons, errContractExpired.Error())
-		renew = false
-		refresh = false
+		usable = false
 		recoverable = false
+		refresh = false
+		renew = false
 	} else if c.Revision.RevisionNumber == math.MaxUint64 {
 		reasons = append(reasons, errContractMaxRevisionNumber.Error())
-		renew = false
-		refresh = false
+		usable = false
 		recoverable = false
+		refresh = false
+		renew = false
 	} else {
 		if isOutOfCollateral(c, s, renterFunds, bh) {
 			reasons = append(reasons, errContractOutOfCollateral.Error())
-			renew = false
-			refresh = true
+			usable = false
 			recoverable = true
+			refresh = true
+			renew = false
 		}
 		if isOutOfFunds(cfg, s, c) {
 			reasons = append(reasons, errContractOutOfFunds.Error())
-			renew = false
-			refresh = true
+			usable = false
 			recoverable = true
+			refresh = true
+			renew = false
 		}
 		if shouldRenew, secondHalf := isUpForRenewal(cfg, *c.Revision, bh); shouldRenew {
-			if secondHalf {
-				reasons = append(reasons, errContractUpForRenewal.Error()) // only unusable if in second half of renew window
-			}
-			renew = true
-			refresh = false
+			reasons = append(reasons, fmt.Errorf("%w; second half: %t", errContractUpForRenewal, secondHalf).Error())
+			usable = !secondHalf // only unusable if in second half of renew window
```

Comment:
Shouldn't this be `usable = usable && !secondHalf`?

Reply:
<3